### PR TITLE
Add virtual cluster api for k8s rest server

### PR DESCRIPTION
**Pull request by abuccts**
Add virtual cluster api for k8s rest server.

### CODE REVIEW

File: src/rest-server/src/models/v2/virtual-cluster/yarn.js [hunk 1]
```diff
@@ -0,0 +1,477 @@
+// Copyright (c) Microsoft Corporation
```
Comment (mzmssg) [inline src/rest-server/src/models/v2/virtual-cluster/yarn.js]: use git mv to keep history?

File: src/rest-server/src/models/v2/virtual-cluster/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+const launcherConfig = require('@pai/config/launcher');
+
+if (launcherConfig.type === 'yarn') {
+  module.exports = require('@pai/models/v2/virtual-cluster/yarn');
+} else {
+  module.exports = require('@pai/models/v2/virtual-cluster/yarn');
```
Comment (mzmssg) [inline src/rest-server/src/models/v2/virtual-cluster/index.js]: change to k8s?

File: src/rest-server/src/models/v2/virtual-cluster/yarn.js (hunk 1 above)
Comment (abuccts) [reply src/rest-server/src/models/v2/virtual-cluster/yarn.js]: this file has been modified

File: src/rest-server/src/models/v2/virtual-cluster/index.js (hunk 2 above)
Comment (abuccts) [reply src/rest-server/src/models/v2/virtual-cluster/index.js]: hasn't implemented k8s vc yet

**Pull request merged**